lightpath/path.py

```diff
@@ -53,14 +85,27 @@ def find_device_state(device):
     # Gather device information
     try:
         _in, _out = device.inserted, device.removed
+        logger.debug("Device %s reporting; IN=%s, OUT=%s",
+                     device.name, _in, _out)
     except Exception as exc:
+        # Catch DisconnectionError
+        if isinstance(exc, DisconnectedError):
+            logger.warning("Unable to connect to %r", device)
+            return DeviceState.Disconnected
+        # Check if this was an error with an EPICS connection
         logger.exception("Unable to determine device state for %r", device)
-        return DeviceState.Faulted
+        return DeviceState.Error
```

Comment:
Should probably add `TimeoutError` too in case this happens in a signal `get` call, which will try to wait for connection.

The canonical way to check for different error types is:
```
except (DisconnectedError, TimeoutError):
    <stuff>
except Exception:
    <stuff>
```
Rather than calling `isinstance` yourself
```